### PR TITLE
feat(connected-overlay): support all overlay config properties

### PR DESCRIPTION
**Pull request by kara**
This PR passes all the overlay config properties through to the connected overlay directive.

r: @jelbourn

### CODE REVIEW

File: src/lib/core/overlay/overlay-directives.ts [hunk 1]
```diff
@@ -50,15 +52,51 @@ export class OverlayOrigin {
...
+  @Input() backdropClass: string;
+
+  /** Whether or not the overlay should attach a backdrop. */
+  @Input()
+  get hasBackdrop() {
+    return this._hasBackdrop;
+  }
+
+  set hasBackdrop(value: any) {
+    this._hasBackdrop = value === '' || (!!value && value !== 'false');
```
Comment (jelbourn) [inline src/lib/core/overlay/overlay-directives.ts]: Should this be the same logic as here?
https://github.com/jelbourn/material2/blob/a70e3e569d026aaf4c0ad8728d9b6091f8673901/src/lib/core/coersion/boolean-property.ts#L3

File: src/lib/core/overlay/overlay-directives.ts [hunk 2]
```diff
@@ -50,15 +52,51 @@ export class OverlayOrigin {
...
+    return this._open;
+  }
+
+  set open(value: boolean) {
+    value ? this._attachOverlay() : this._detachOverlay();
+    this._open = value;
+  }
+
+  /** Event emitted when the backdrop is clicked. */
+  @Output() backdropClick = new EventEmitter();
```
Comment (jelbourn) [inline src/lib/core/overlay/overlay-directives.ts]: Doesn't this require a generic type?

Comment (kara) [inline src/lib/core/overlay/overlay-directives.ts]: Yeah, I think it does. It's not throwing, so maybe it's implicitly typed to `EventEmitter<void>` when set this way?

File: src/lib/core/overlay/overlay-directives.ts (hunk 1 above)
Comment (jelbourn) [inline src/lib/core/overlay/overlay-directives.ts]: Not in master yet; I'd copy the logic and add a TODO

**Pull request merged**